### ISSUE TITLE
example/invertable_operator doesn't compile

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#1675]
```
Trying to compile ../../../examples/invertable_operator
  Compiling  invertable_operator.cxx
invertable_operator.cxx: In member function ‘virtual int HW::init(bool)’:
invertable_operator.cxx:87:14: error: ‘class bout::inversion::InvertableOperator<Field3D>’ has no member named ‘setOperatorFunction’
     mySolver.setOperatorFunction(mm);
              ^~~~~~~~~~~~~~~~~~~
invertable_operator.cxx:88:14: error: ‘class bout::inversion::InvertableOperator<Field3D>’ has no member named ‘setup’
     mySolver.setup();
              ^~~~~
invertable_operator.cxx:75:17: warning: unused parameter ‘restart’ [-Wunused-parameter]
   int init(bool restart) {
            ~~~~~^~~~~~~
invertable_operator.cxx: In member function ‘virtual int HW::rhs(BoutReal)’:
invertable_operator.cxx:106:28: error: ‘class bout::inversion::InvertableOperator<Field3D>’ has no member named ‘invert’
     solutionInv = mySolver.invert(n, 0.0);
                            ^~~~~~
invertable_operator.cxx:107:14: error: ‘class bout::inversion::InvertableOperator<Field3D>’ has no member named ‘reportTime’
     mySolver.reportTime();
              ^~~~~~~~~~
invertable_operator.cxx:112:32: error: ‘class bout::inversion::InvertableOperator<Field3D>’ has no member named ‘invert’
         solutionInv = mySolver.invert(n, solutionInv);
                                ^~~~~~
invertable_operator.cxx:121:26: error: ‘class bout::inversion::InvertableOperator<Field3D>’ has no member named ‘verify’
     auto pass = mySolver.verify(n) == 0 ? "False" : "True";
                          ^~~~~~
invertable_operator.cxx:139:14: error: ‘class bout::inversion::InvertableOperator<Field3D>’ has no member named ‘reportTime’
     mySolver.reportTime();
              ^~~~~~~~~~
invertable_operator.cxx:100:20: warning: unused parameter ‘time’ [-Wunused-parameter]
   int rhs(BoutReal time) {
           ~~~~~~~~~^~~~
make: *** [../../make.config:308: invertable_operator.o] Error 1
invertable_operator failed
```

**Comment by d7919** [comment]
The invertable operator class is only available in BOUT++ builds configured with PETSc support -- I'm guessing this case doesn't have PETSc?

**Comment by dschwoerer** [comment]
Yes, I am first running all tests that claim to not require petsc, and later run all tests that require petsc.

I will split out the example in the tests.

Thanks for clarifying.

**Comment by d7919** [comment]
Yes, I guess it might be nice to "port" the requires checking to the examples as well.

Also I suppose I should have probably made non-PETSc builds provide an empty version of the invertable operator class that throws with a helpful error message if built without PETSc.

**Comment by dschwoerer** [comment]
I think failing to build is in this case fine, as that way I noticed, and can make sure it compiles with the different petsc versions.

I try to put it in the makefile ...

**Comment by dschwoerer** [comment]
#1678 should resolve this

**Comment by dschwoerer** [comment]
fixed via #1678
